datashuttle/tui/app.py

```diff
@@ -0,0 +1,72 @@
+from pathlib import Path
+from time import monotonic
+
+from textual.app import App, ComposeResult
+from textual.binding import Binding
+from textual.widgets import (
+    DirectoryTree,
+    Footer,
+    Header,
+    Input,
+    Label,
+    TabbedContent,
+    TabPane,
+)
+
+
+class TuiApp(App):
+    """
+    The main app page for the DataShuttle TUI.
+
+    Running this application in a main block as below
+    if __name__ == __main__:
+         app = MyApp()
+         app.run()
+
+    Initialises the TUI event loop and starts the application.
+    """
+
+    TITLE = "DataShuttle"
+
+    # Set path to TCSS
+    tui_path = Path(__file__).parent
+    CSS_PATH = tui_path / "css" / "tab_content.tcss"
+
+    # Set key-bindings
+    BINDINGS = [
+        Binding("ctrl+d", "toggle_dark", "Toggle Dark Mode", priority=True)
+    ]
+
+    # Compose window
+    def compose(self) -> ComposeResult:
+        yield Header()
+        with TabbedContent():
+            with TabPane("Create", id="create"):
+                yield DirectoryTree(str(Path().home()), id="FileTree")
+                yield Label("Folder Name")
+                yield Input(
+                    placeholder="Double-click on any folder to fill this field.",
+                    id="subject",
+                )
+            with TabPane("Transfer", id="transfer"):
+                yield Label("Transfer; Seems to work!")
+        yield Footer()
+
+    prev_time = 0.0
+
+    def on_directory_tree_directory_selected(
+        self, event: DirectoryTree.DirectorySelected
+    ):
+        """
+        After double-clicking a directory within the directory-tree widget, replaces contents of the \'Subject\'
```

Comment:
```suggestion
        After double-clicking a directory within the directory-tree 
        widget, replaces contents of the 'Subject' input widget 
        with directory name.
```

Reply:
Just to keep within line limits (Black does not format strings unfortunately). I have previoulsy been referrring to the textual widgets with capital leter e.g. Input but, if you agree I think lowercase reads better in the end.